ambry-api/src/test/java/com.github.ambry/account/AccountContainerTest.java

```diff
@@ -0,0 +1,574 @@
+      assertContainer(containerFromBuilder, i);
+
+      // build a container from existing container
+      containerBuilder = new ContainerBuilder(containerFromBuilder);
+      Container container2ByBuilder = containerBuilder.build();
+      assertContainer(container2ByBuilder, i);
+    }
+  }
+
+  /**
+   * Tests required fields are missing to build an account.
+   */
+  @Test
+  public void testFieldMissingToBuildAccount() {
+    // test when required fields are null
+    assertBuildAccountWithMissingFields(null, refAccountName, refAccountStatus, IllegalStateException.class);
+    assertBuildAccountWithMissingFields(refAccountId, null, refAccountStatus, IllegalStateException.class);
+    assertBuildAccountWithMissingFields(refAccountId, refAccountName, null, IllegalStateException.class);
+  }
+
+  /**
+   * Tests required fields are missing to build an account.
+   */
+  @Test
+  public void testFieldMissingToBuildContainer() {
+    // test when required fields are null
+    assertBuildContainerWithMissingFields(null, refContainerNames.get(0), refContainerStatuses.get(0),
+        refContainerPrivacies.get(0), refAccountId, IllegalStateException.class);
+    assertBuildContainerWithMissingFields(refContainerIds.get(0), null, refContainerStatuses.get(0),
+        refContainerPrivacies.get(0), refAccountId, IllegalStateException.class);
+    assertBuildContainerWithMissingFields(refContainerIds.get(0), refContainerNames.get(0), null,
+        refContainerPrivacies.get(0), refAccountId, IllegalStateException.class);
+    assertBuildContainerWithMissingFields(refContainerIds.get(0), refContainerNames.get(0), refContainerStatuses.get(0),
+        null, refAccountId, IllegalStateException.class);
+    assertBuildContainerWithMissingFields(refContainerIds.get(0), refContainerNames.get(0), refContainerStatuses.get(0),
+        refContainerPrivacies.get(0), null, IllegalStateException.class);
+  }
+
+  /**
+   * Tests removing containers from an account, and then update the account.
+   * @throws JSONException
+   */
+  @Test
+  public void testRemoveContainerAndUpdateAccount() throws JSONException {
+    // set an account with different field value
+    Account origin = Account.fromJson(accountJsonLike);
+    AccountBuilder accountBuilder = new AccountBuilder(origin);
+    short updatedAccountId = refAccountId + 1;
+    String updatedAccountName = refAccountName + "-updated";
+    Account.AccountStatus updatedAccountStatus = Account.AccountStatus.INACTIVE;
+    accountBuilder.setId(updatedAccountId);
+    accountBuilder.setName(updatedAccountName);
+    accountBuilder.setStatus(updatedAccountStatus);
+
+    try {
+      accountBuilder.build();
+    } catch (IllegalStateException e) {
+      // expected, as new account id does not match the parentAccountId of the two containers.
+      System.out.println(e.getMessage());
+    }
+
+    // remove two existing containers.
+    for (Container container : origin.getAllContainers()) {
```

Comment:
is there a test, wherein you remove just one container and test if the other exists. Ideally, if you have N containers, remove random M (<n) containers and validate. Also remove all N and validate.